client/my-sites/earn/memberships/products.jsx

```diff
@@ -94,60 +88,30 @@ class MembershipsProductsSection extends Component {
 						{ this.renderEllipsisMenu( product.ID ) }
 					</CompactCard>
 				) ) }
-				{ this.state.showDialog && (
+				{ this.state.showAddEditDialog && (
 					<RecurringPaymentsPlanAddEditModal
 						closeDialog={ this.closeDialog }
 						product={ this.state.product }
 					/>
 				) }
-				<Dialog
-					isVisible={ !! this.state.deletedProductId }
-					buttons={ [
-						{
-							label: this.props.translate( 'Cancel' ),
-							action: 'cancel',
-						},
-						{
-							label: this.props.translate( 'Delete' ),
-							isPrimary: true,
-							action: 'delete',
-						},
-					] }
-					onClose={ this.onCloseDeleteProduct }
-				>
-					<h1>{ this.props.translate( 'Confirmation' ) }</h1>
-					<p>
-						{ this.props.translate( 'Do you want to delete "%s"?', {
-							args: get(
-								this.props.products.filter( ( p ) => p.ID === this.state.deletedProductId ),
-								[ 0, 'title' ],
-								''
-							),
-						} ) }
-					</p>
-					<Notice
-						text={ this.props.translate(
-							'Deleting a product does not cancel the subscription for existing subscribers.{{br/}}They will continue to be charged even after you delete it.',
-							{ components: { br: <br /> } }
-						) }
-						showDismiss={ false }
+				{ this.state.showDeleteDialog && (
+					<RecurringPaymentsPlanDeleteModal
+						closeDialog={ this.closeDialog }
+						product={ this.state.product }
 					/>
-				</Dialog>
+				) }
 			</div>
 		);
 	}
 }
 
-export default connect(
-	( state ) => {
-		const site = getSelectedSite( state );
-		const siteId = getSelectedSiteId( state );
-		return {
-			site,
-			siteId,
-			siteSlug: getSelectedSiteSlug( state ),
-			products: get( state, [ 'memberships', 'productList', 'items', siteId ], [] ),
-		};
-	},
-	{ requestDeleteProduct }
-)( localize( MembershipsProductsSection ) );
+export default connect( ( state ) => {
+	const site = getSelectedSite( state );
+	const siteId = getSelectedSiteId( state );
+	return {
+		site,
+		siteId,
+		siteSlug: getSelectedSiteSlug( state ),
+		products: get( state, [ 'memberships', 'productList', 'items', siteId ], [] ),
```

Comment:
Optional, but if we're tidying up things in this file we can remove the lodash get usage in favor of optional chaining:

```jsx
products: state?.memberships?.productList?.items?.[ siteId ]
```

Reply:
Nice catch: 07607bba7e